tests/phpunit/FileSystem/SourceFileFilterTest.php

```diff
@@ -0,0 +1,140 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\FileSystem;
+
+use Generator;
+use Infection\FileSystem\SourceFileFilter;
+use IteratorIterator;
+use PHPUnit\Framework\TestCase;
+use function Pipeline\take;
+use Symfony\Component\Finder\SplFileInfo;
+use Traversable;
+
+/**
+ * @covers \Infection\FileSystem\SourceFileFilter
+ */
+final class SourceFileFilterTest extends TestCase
+{
+    public function test_it_parses_empty_filters(): void
+    {
+        $fileFilter = new SourceFileFilter('');
+        $this->assertSame([], $fileFilter->getFilters());
+    }
+
+    public function test_it_parses_filters(): void
+    {
+        $fileFilter = new SourceFileFilter('src/Foo.php, src/Bar.php');
+        $this->assertSame([
+            'src/Foo.php',
+            'src/Bar.php',
+        ], $fileFilter->getFilters());
+    }
+
+    /**
+     * @dataProvider fileListProvider
+     */
+    public function test_it_filters_traversable(string $filter, array $input, array $expected): void
+    {
+        $input = self::arrayToSplFileInfoTraversable($input);
+
+        $this->assertFilter($filter, $input, $expected);
+    }
+
+    /**
+     * @dataProvider fileListProvider
+     */
+    public function test_it_filters_iterator(string $filter, array $input, array $expected): void
+    {
+        $input = self::arrayToSplFileInfoTraversable($input);
+
+        $input = new IteratorIterator($input);
+
+        $this->assertFilter($filter, $input, $expected);
+    }
+
+    public static function fileListProvider(): Generator
+    {
+        yield ['src/Example', ['src/Example/Test.php'], ['src/Example/Test.php']];
+
+        yield ['src/Foo', ['src/Example/Test.php'], []];
+
+        yield ['', [
+            'src/Foo/Test.php',
+            'src/Bar/Baz.php',
+            'src/Example/Test.php',
+        ], [
+            'src/Foo/Test.php',
+            'src/Bar/Baz.php',
+            'src/Example/Test.php',
+        ]];
+
+        yield ['src/Foo,src/Bar', [
+            'src/Foo/Test.php',
+            'src/Bar/Baz.php',
+            'src/Example/Test.php',
+        ], [
+            'src/Foo/Test.php',
+            'src/Bar/Baz.php',
+        ]];
+    }
+
+    private function assertFilter(string $filter, iterable $input, array $expected): void
+    {
+        $fileFilter = new SourceFileFilter($filter);
+        $actual = $fileFilter->filter($input);
+
+        $actual = take($actual)
+            ->map(static function (SplFileInfo $fileInfo) {
+                return $fileInfo->getRealPath();
+            })
+            ->toArray();
+
+        $this->assertSame($expected, $actual);
+    }
+
+    private function arrayToSplFileInfoTraversable(array $input): Traversable
```

Comment:
Maybe I misunderstood but isn't the downstream API requiring an `iterable` here in which case we could just do a simple `array_map` and return an `array` instead?

Reply:
Yes, here we can see that downstream can take not only arrays, but interables in general.